Src/Utils/Paths.ts

```diff
@@ -1,6 +1,14 @@
+import finder from "find-package-json";
 import path from "path";
 
-export const ROOT_DIR: string = path.normalize(`${__dirname}/../../`);
+// crawl up parent directories until we find the package.json of the calling software
+let lRootDir: string = process.cwd();
+for (const lPackageJson of finder())
+{
+    lRootDir = path.dirname(lPackageJson.__path);
```

Comment:
Have you tested how far up this will crawl, i assume it will keep going until it reaches `/`? This _should_ be fine but a better solution may be to crawl upwards until you find `Gulpfile.ts`/`js`?

Reply:
So I tested this by starting the crawl 3 levels inside a package in the node_modules/ and it worked without any issues. I've also tested by running gulp at Src/Utils/ and it worked just fine.

Have tested on typescript-template as well.

Reply:
But you are not guaranteed to find a `Gulpfile.ts` at the same location as a `tsconfig.json` right? by definition if you crawl to the first `Gulpfile.ts` you will be guaranteed to have found a Gulpfile. Implementing this crawler should not be to difficult, just start at `process.cwd` and take off a dir at a time until the dir contains a Gulpfile?

Reply:
done!
![image](https://user-images.githubusercontent.com/2607235/152483852-db2cf456-67df-4aa6-a5df-5fcb66ecc00b.png)